### PR TITLE
workflows: retry while managing record via hep

### PR DESCRIPTION
**Pull request by harunurhan**
* Retries when inspirehep returns bad gateway error possibly due to
network issues.

### CODE REVIEW

**Comment by drjova** [comment]
@harunurhan you can rebase now

File: inspirehep/modules/workflows/tasks/upload.py [hunk 1]
```diff
@@ -143,13 +146,15 @@ def send_record_to_hep(obj, endpoint, control_number=None):
...
+        raise BadGatewayError()
+
     try:
         error_msg = response.json()
     except JSONDecodeError:
         error_msg = response.text
     raise WorkflowsError(
-        "Response code from Inspirehep is {code}."
-        " This is wrong! Message from the server: {message}".format(
+        "Error from inspirehep [${code}]: {message}".format(
```
Comment (drjova) [inline inspirehep/modules/workflows/tasks/upload.py]: python :P

File: tests/integration/workflows/test_workflows_tasks_upload.py [hunk 2]
```diff
@@ -359,6 +359,44 @@ def test_store_record_inspirehep_api_author_update(workflow_app):
...
+                        'uuid': expected_head_uuid
+                    }}
+                ],
+            )
+            store_record(workflow, eng)
+
+    assert workflow.data['control_number'] == expected_control_number
+    assert workflow.extra_data['head_uuid'] == expected_head_uuid
+
+
```
Comment (drjova) [inline tests/integration/workflows/test_workflows_tasks_upload.py]: can we add a test with the update? ``200``

Comment (harunurhan) [reply tests/integration/workflows/test_workflows_tasks_upload.py]: there is already one

**Pull request merged**